astropy/cosmology/core.py

```diff
@@ -66,11 +202,37 @@ class Cosmology(metaclass=ABCMeta):
     read = UnifiedReadWriteMethod(CosmologyRead)
     write = UnifiedReadWriteMethod(CosmologyWrite)
 
+    # Parameters
+    __parameters__ = ()
+    _order_params_by_signature = False
```

Comment:
I'm wondering whether this is useful - is there actually a class in which it should not be `True`? Why not always reorder?

Reply:
For the current set of cosmologies, no reason. I just thought it might result in unexpected behavior for users. But since this is a new behavior, I guess there's no real expectation one way or the other! SGTM to make it True by default.

My thought for allowing the reordering to be enabled/disabled comes from looking at ``astropy.model.Model``, where the init signature need not match the parameters (though it usually does). Similarly, it's not unreasonable to define a Cosmology where with ``h`` as the input argument, but ``H0`` as the parameter. In this case, the H0 parameter should still be first but the reorder-by-signature would get this wrong.

Reply:
OK, fair enough. Though perhaps an argument that the parameters should not be by default inferred from `__init__`, but rather from the `Parameter`s given. (Obviously, if such a change is made, much more logical in a follow-up PR!)

Reply:
However, I do think it's reasonable to push this to a future PR, if anyone ever actually needs it. Now that I now it's easy to do, it'll be easy to redo.

Reply:
Resolved. Pushed to later. Thanks!